packages/plugin/src/generators/e2e-project/e2e.ts

```diff
@@ -1,50 +1,85 @@
-import type { Tree } from '@nx/devkit';
 import {
   addProjectConfiguration,
-  extractLayoutDirectory,
   formatFiles,
   generateFiles,
-  GeneratorCallback,
   getPackageManagerCommand,
-  getWorkspaceLayout,
   joinPathFragments,
   names,
   offsetFromRoot,
   readJson,
   readNxJson,
   readProjectConfiguration,
   runTasksInSerial,
+  updateJson,
   updateProjectConfiguration,
+  writeJson,
+  type GeneratorCallback,
+  type ProjectConfiguration,
+  type Tree,
 } from '@nx/devkit';
-import { determineProjectNameAndRootOptions } from '@nx/devkit/src/generators/project-name-and-root-utils';
+import {
+  determineProjectNameAndRootOptions,
+  resolveImportPath,
+} from '@nx/devkit/src/generators/project-name-and-root-utils';
+import { promptWhenInteractive } from '@nx/devkit/src/generators/prompt';
+import { LinterType, lintProjectGenerator } from '@nx/eslint';
 import { addPropertyToJestConfig, configurationGenerator } from '@nx/jest';
 import { getRelativePathToRootTsConfig } from '@nx/js';
 import { setupVerdaccio } from '@nx/js/src/generators/setup-verdaccio/generator';
 import { addLocalRegistryScripts } from '@nx/js/src/utils/add-local-registry-scripts';
-import { assertNotUsingTsSolutionSetup } from '@nx/js/src/utils/typescript/ts-solution-setup';
-import { Linter, LinterType, lintProjectGenerator } from '@nx/eslint';
+import {
+  getProjectPackageManagerWorkspaceState,
+  getProjectPackageManagerWorkspaceStateWarningTask,
+} from '@nx/js/src/utils/package-manager-workspaces';
+import { isUsingTsSolutionSetup } from '@nx/js/src/utils/typescript/ts-solution-setup';
+import type { PackageJson } from 'nx/src/utils/package-json';
 import { join } from 'path';
 import type { Schema } from './schema';
 
 interface NormalizedSchema extends Schema {
   projectRoot: string;
   projectName: string;
   pluginPropertyName: string;
-  linter: Linter | LinterType;
+  linter: LinterType;
+  useProjectJson: boolean;
+  addPlugin: boolean;
+  isTsSolutionSetup: boolean;
 }
 
 async function normalizeOptions(
   host: Tree,
   options: Schema
 ): Promise<NormalizedSchema> {
+  const isTsSolutionSetup = isUsingTsSolutionSetup(host);
+
+  let linter = options.linter;
```

Comment:
We do these prompts a lot, maybe it's a clean up item later to extract a shared util to `@nx/js`.